### PR TITLE
update for ts26 strict

### PR DESCRIPTION
**Pull request by scottbessler**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -9,8 +9,6 @@ dependencies:
 test:
   pre:
     - tsc
-  post:
-    - mv coverage/* $CIRCLE_ARTIFACTS/ || exit_on_error "No coverage!"
```
Comment (mschoendorf) [inline circle.yml]: no codecov?

Comment (scottbessler) [reply circle.yml]: first prio was having something working, given the size/exposure of the repo

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        descriptor: TypedPropertyDescriptor<any>
+    ) => {
+        let functionName: string = String(propertyName);
+
+        const className = target.constructor.name;
+        let method = descriptor.value;
+        descriptor.value = function() {
+            let start = new Date();
+            if (!method) {
+                throw new Error('method missing');
```
Comment (mschoendorf) [inline src/index.ts]: do we want to throw in here? could make a function fail that we wouldn't want to fail due to datadog, and i don't know how to wrap an annotation in a try catch.

File: tsconfig.json [hunk 3]
```diff
@@ -5,12 +5,13 @@
     "target": "es6",
     "sourceMap": true,
     "experimentalDecorators": true,
-    "strictNullChecks": true
+    "strictNullChecks": true,
+    "strict": true
```
Comment (mschoendorf) [inline tsconfig.json]: noUnused*?

**Review by mschoendorf — APPROVED** [review]

File: src/index.ts (hunk 2 above)
Comment (scottbessler) [reply src/index.ts]: tbh, i didnt want to break anything so left it alone.

**Pull request merged**